src/AstProfileUse.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2018, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstProfileUse.h
+ *
+ * Defines a simple class to query profile data from a profile
+ * for profile-guided optimisation.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "AstAnalysis.h"
+#include "AstRelation.h"
+#include "profile/ProgramRun.h"
+#include "profile/Reader.h"
```

Comment:
Remove AstRelation.h, profile/Reader.h, map, set, stack, vector, and I think utility. Add memory

Reply:
Fixed.